### PR TITLE
fix(chore): remove processing and waiting queue

### LINKED ISSUE #20664: chore(gatsby): processing queue is never set with a job id
## Summary

`processing` queue is never set with a job id

## Description

in this line the queue `processing` is defined:

https://github.com/gatsbyjs/gatsby/blob/631ba49765463ab514336a84b9182ec3600c76c1/packages/gatsby/src/query/queue.js#L26

and it is checked i line 45:

https://github.com/gatsbyjs/gatsby/blob/631ba49765463ab514336a84b9182ec3600c76c1/packages/gatsby/src/query/queue.js#L45-L50

and jobs deleted in 

https://github.com/gatsbyjs/gatsby/blob/631ba49765463ab514336a84b9182ec3600c76c1/packages/gatsby/src/query/queue.js#L69

but in this file i can not find a `processing.add()`, maybe there is some hidden place where this is set?

i assume there is a missing command in line 49
```diff
 if (processing.has(job.id)) { 
   waiting.set(job.id, job) 
   cb(`already running`) 
 } else { 
+  processing.add(job.id)
   cb(null, job) 
 } 
   
```

i like to cc @pvdz @vladar for confirmation

### PR DESCRIPTION
**Pull request by muescha**
## Description

remove processing queue and waiting queue

## Related Issues

fixes #20664 chore(gatsby): processing queue is never set with a job id

### CODE REVIEW

File: packages/gatsby/src/query/queue.js [hunk 1]
```diff
@@ -39,15 +35,10 @@ const createDevelopQueue = getRunner => {
...
+    // When the query finishes, it checks the waiting map and pushes
     // another job to make sure all the user changes are captured.
     filter: (job, cb) => {
-      if (processing.has(job.id)) {
-        waiting.set(job.id, job)
-        cb(`already running`)
-      } else {
-        cb(null, job)
-      }
+      cb(null, job)
```
Comment (vladar) [inline packages/gatsby/src/query/queue.js]: I think we don't need `filter` at all anymore (as it basically doesn't filter anything).

Comment (pieh) [inline packages/gatsby/src/query/queue.js]: this seems like no-op right now, and I think we can remove `filter` completely (?)

Comment (pieh) [reply packages/gatsby/src/query/queue.js]: LOL, @vladar left same comment, so I'll just resolve mine as duplicate ;)

**Comment by muescha** [comment]
@sidharthachatterjee thx for the fix to get it merged

**Pull request merged**